File: container-with-most-water/container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        maxArea = 0

        left = 0
        right = len(height)-1

        while left < right:
            width = right - left
            testArea = (height[right] * width) if height[left] >= height[right] else (height[left] * width)

            if (testArea > maxArea):
                maxArea = testArea

            if height[left] < height[right]:
                left += 1
        
            elif height[left] >= height[right]:
                right -= 1

        return maxArea

File: container-with-most-water/container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    var maxArea = 0

    var left = 0
    var right = height.length-1

    while (left < right) {
        var width = right - left
        var leftHeight = height[left]
        var rightHeight = height[right]

        var testArea = (leftHeight > rightHeight) ? rightHeight * width : leftHeight * width

        if (testArea > maxArea) {
            maxArea = testArea
        }

        if (leftHeight > rightHeight) {
            right -= 1
        } else if (leftHeight <= rightHeight) {
            left += 1
        }
    }

    return maxArea
};